src/plugins/chart_expressions/expression_metric/public/components/metric_vis.test.tsx

```diff
@@ -1124,7 +1125,7 @@ describe('MetricVisComponent', function () {
         value: primaryMetric,
         valueFormatter,
         extra,
-      } = component.find(Metric).props().data?.[0][0]!;
+      } = component.find(Metric).props().data?.[0][0]! as MetricWNumber;
 
       return { primary: valueFormatter(primaryMetric), secondary: extra?.props.children[1] };
```

Comment:
@andrewctate I've took a look at this test and it doesn't seems to test at all the behaviour of the metric chart. It instead just picks the props values/functions and apply the passed formatted to the value. This doesn't test how the value formatter and the value itself are rendered by the metric component.
It will be better if you can mount the component and check the HTML output